src/components/inline_edit/inline_edit_form.tsx

```diff
@@ -168,8 +187,16 @@ export const EuiInlineEditForm: FunctionComponent<EuiInlineEditFormProps> = ({
   ]);
 
   const [isEditing, setIsEditing] = useState(false || startWithEditOpen);
-  const [editModeValue, setEditModeValue] = useState(defaultValue);
-  const [readModeValue, setReadModeValue] = useState(defaultValue);
+  const [editModeValue, setEditModeValue] = useState(value || defaultValue);
+  const [readModeValue, setReadModeValue] = useState(value || defaultValue);
```

Comment:
My 2c is I don't think we need to listen the `value ||` here for our internal state. Since they're only used for uncontrolled behavior, they only need to listen for `defaultValue`.

```suggestion
  const [editModeValue, setEditModeValue] = useState(defaultValue);
  const [readModeValue, setReadModeValue] = useState(defaultValue);
```

Reply:
This logic is actually how I'm able to pass `readModeValue` to the `onCancel` function. Otherwise, the `readModeValue` falls out of sync and isn't a reliable backup of the previous value.
Link to code sample mentioned: https://github.com/elastic/eui/pull/7117#discussion_r1303414143

Reply:
If readModeValue is the only one that needs this logic, let's set it just for that and add an inline comment noting why